src/lib/commandBarConfigs/validators.ts

```diff
@@ -153,3 +153,71 @@ export const loftValidator = async ({
     return 'Unable to loft with selected sketches'
   }
 }
+
+export const shellValidator = async ({
+  data,
+  context,
+}: {
+  data: any
+  context: CommandBarContext
+}): Promise<boolean | string> => {
+  if (!isSelections(data.selection)) {
+    return 'Unable to revolve, selections are missing'
+  }
+  const selection = data.selection as Selections
+  console.log('selection', selection)
+  const firstArtifact = selection.graphSelections[0].artifact
+
+  if (!firstArtifact) {
+    return 'Unable to shell, no artifact found'
+  }
+
+  if (!(firstArtifact.type === 'cap' || firstArtifact.type === 'wall')) {
+    return 'Unable to shell, first artifact is not a cap or a wall'
+  }
+
+  console.log('selection artifact', firstArtifact)
+  console.log('artifactGraph', engineCommandManager.artifactGraph)
+  const sweep = engineCommandManager.artifactGraph.get(firstArtifact.sweepId)
+  if (!sweep || sweep?.type !== 'sweep') {
+    return 'Unable to shell, couldnt find pathId'
+  }
+
+  const faceId = firstArtifact.id
+  const hasOtherObjectIds = selection.graphSelections.some(
+    (s) =>
+      (s.artifact?.type === 'cap' || s.artifact?.type === 'wall') &&
+      s.artifact.sweepId !== firstArtifact.sweepId
+  )
+  if (hasOtherObjectIds) {
+    return 'Unable to shell, selection is across solids'
+  }
+
+  // TODO: NOT WORKING YET
```

Comment:
After talking to the engine team shell isn't supported yet. Work started at https://github.com/KittyCAD/engine/tree/ben/shell_dry_run, once merged I'll try again and get this ready to go